pkg/api/controller_test.go

```diff
@@ -1502,11 +1502,22 @@ func TestController_ListBranchesHandler(t *testing.T) {
 		if len(results) != 2 {
 			t.Fatalf("expected 2 branches to return, got %d", len(results))
 		}
-		retReference := results[0]
-		const expectedID = "main2"
-		if retReference.Id != expectedID {
-			t.Fatalf("expected '%s' as the first result for the second page, got '%s' instead", expectedID, retReference.Id)
+		require.Equal(t, results[0].Id, "main3")
+		require.Equal(t, results[1].Id, "main5")
+
+		// List all branches
+		resp, err = clt.ListBranchesWithResponse(ctx, repo, &apigen.ListBranchesParams{
+			After:      apiutil.Ptr[apigen.PaginationAfter]("main1"),
+			Amount:     apiutil.Ptr[apigen.PaginationAmount](2),
+			ShowHidden: swag.Bool(true),
+		})
+		verifyResponseOK(t, resp, err)
+		results = resp.JSON200.Results
+		if len(results) != 2 {
+			t.Fatalf("expected 2 branches to return, got %d", len(results))
 		}
+		require.Equal(t, results[0].Id, "main2")
+		require.Equal(t, results[1].Id, "main3")
```

Comment:
I prefer to compare outputs directly, it makes debugging easier when there are any failures.  So in this case I would extract `.Id`s to a slice, and then use a contents comparison such as deep.Equal (or perhaps testify has a usable slice equality test).

Reply:
I'm sorry - but this feels like a complete overkill.
We're comparing strings, why would we need deep equals? Also I don't understand how putting the expected and received in a slice and comparing slices makes it easier to debug?
These tests run in ms, so putting a breakpoint where you need and looking and the variables takes less than a second.

Reply:
Debuggers are the antithesis of testing.
The advantage of comparing slices of strings correctly is that you see entire context.  

- For instance, suppose you receive ["main3"].    You get the useless error message "expected 2 branches to return", and not even the branches you did get.
- For instance, suppose you receive ["main3", "main2"] (maybe some KV weirdness).  You get the useless error message "got main3 expected main2" which still doesn't say what happened.
- For instance, suppose you receive ["main1", "main2"] (some filtering is broken).  You get the useless error message "got main1 expected main2" which still doesn't say what happened.

Tests should fail with a useful error message, if only to help understand how much work is required to fix.

Reply:
modified